packages/fast-components-react-base/src/tabs/tabs.tsx

```diff
@@ -0,0 +1,336 @@
+import { ITabsHandledProps, ITabsManagedClasses, ITabsUnhandledProps, TabsProps } from "./tabs.props";
+
+export enum TabSlot {
+    tab = "tab",
+    tabItem = "tab-item",
+    tabPanel = "tab-panel"
+}
+
+export interface ITabsState {
+    activeId: string;
+}
+
+class Tabs extends Foundation<ITabsHandledProps & ITabsManagedClasses, ITabsUnhandledProps, ITabsState> {
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(nextProps: TabsProps, prevState: ITabsState): null | ITabsState {
+        if (nextProps.activeId && nextProps.activeId !== prevState.activeId) {
+            return {
+                activeId: nextProps.activeId
+            };
+        }
+
+        return null;
+    }
+
+    protected handledProps: HandledProps<ITabsHandledProps & IManagedClasses<ITabsClassNameContract>> = {
+        activeId: void 0,
+        children: void 0,
+        label: void 0,
+        managedClasses: void 0,
+        orientation: void 0,
+        tabItemSlot: void 0,
+        tabPanelSlot: void 0,
+        tabSlot: void 0
+    };
+
+    /**
+     * React ref for the tab list
+     */
+    private tabListRef: React.RefObject<HTMLDivElement>;
+
+    constructor(props: TabsProps) {
+        super(props);
+
+        const tabItems: JSX.Element[] = this.getChildBySlot(
+            this.props.children,
+            this.getSlot(TabSlot.tabItem)
+        );
+        this.tabListRef = React.createRef();
+
+        this.state = {
+            activeId: this.props.activeId
+                ? this.props.activeId
+                : tabItems.length > 0
+                ? tabItems[0].props.id
+                : ""
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        const tabElements: JSX.Element[] = this.renderTabElements();
+
+        return (
+            <div
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+            >
+                <div
+                    role="tablist"
+                    ref={this.tabListRef}
+                    className={this.props.managedClasses.tab_list}
+                    aria-label={this.props.label}
+                    aria-orientation={this.props.orientation}
+                >
+                    {tabElements}
+                </div>
+                {this.renderTabPanels()}
+            </div>
+        );
+    }
+
+    /**
+     * Generates class names based on props
+     */
+    protected generateClassNames(): string {
+        return super.generateClassNames(get(this.props, "managedClasses.tabs"));
+    }
+
+    /**
+     * Renders the tab elements
+     */
+    private renderTabElements(): JSX.Element[] {
+        const tabElements: JSX.Element[] = [];
+
+        this.getChildBySlot(
+            this.props.children,
+            this.getSlot(TabSlot.tabItem)
+        ).forEach((tabItem: JSX.Element, index: number): void => {
+            tabElements.push(
```

Comment:
This is pretty nitty but I think we should avoid mutating functions when possible. This operation is safe, but I think reducing mutating operations where we can helps mitigate folks referencing code examples and applying them to new areas where mutating functions may be problematic.

By using `map` instead of `forEach`, you avoid both mutation and creating a variable - you can just return the result of the `map` function.